### PR TITLE
Initial split of BridgeAppSDK into a framework that doesn't tie to BridgeSDK

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
Additional manual testing completed and unit tests have been fixed.

I am keeping the names so the new framework will also be tagged with the `SBA` prefix. I did this so that anyone who is currently using BridgeAppSDK isn't left out in the cold. Sad and alone and totally confused. It was also way less work. ;)

### CODE REVIEW

File: ResearchUXFactory/Info.plist [hunk 1]
```diff
@@ -19,7 +19,7 @@
 	<key>CFBundleSignature</key>
 	<string>????</string>
 	<key>CFBundleVersion</key>
-	<string>10</string>
+	<string>11</string>
```
Comment (Erin-Mounts) [inline ResearchUXFactory/Info.plist:22]: should we start with build 1 for a new repo? not that it matters in the long run

File: ResearchUXFactory/SBABaseSurveyFactory.swift [hunk 2]
```diff
@@ -58,16 +53,83 @@ open class SBASurveyFactory : NSObject, SBASharedInfoController {
...
+        if let sections = dictionary["sections"] as? [NSDictionary] {
+            self.consentDocument.sections = sections.map({ (dictionarySection) -> ORKConsentSection in
+                let consentSection = dictionarySection.createConsentSection(previous: previousSectionType)
+                previousSectionType = dictionarySection.consentSectionType
+                return consentSection
+            })
+        }
+        
+        // Load the document for the HTML content
+        if let properties = dictionary["documentProperties"] as? NSDictionary,
```
Comment (Erin-Mounts) [inline ResearchUXFactory/SBABaseSurveyFactory.swift]: ditto for the assumption that a survey dictionary with an htmlDocument in its documentProperties is a consent flow.

File: ResearchUXFactory/SBABaseSurveyFactory.swift [hunk 3]
```diff
@@ -58,16 +53,83 @@ open class SBASurveyFactory : NSObject, SBASharedInfoController {
     
     public convenience init(dictionary: NSDictionary) {
         self.init()
+        
+        // Load the sections
+        var previousSectionType: SBAConsentSectionType?
```
Comment (Erin-Mounts) [inline ResearchUXFactory/SBABaseSurveyFactory.swift]: It looks like the code assumes if there are sections, the "survey" is actually the consent flow? If so, a comment (or doc comment) to that effect would be good.

File: ResearchUXFactory/Info.plist (hunk 1 above)
Comment (syoung-smallwisdom) [reply ResearchUXFactory/Info.plist:22]: This was purposefully done to make it easier for me to track the build. :)

**Comment by Erin-Mounts** [comment]
Just a couple of questions/suggestions, otherwise looks good--merge when ready

File: ResearchUXFactory/SBABaseSurveyFactory.swift (hunk 3 above)
Comment (syoung-smallwisdom) [reply ResearchUXFactory/SBABaseSurveyFactory.swift]: These were pulled into the base class b/c I couldn't support multiple inheritance in the default override class defined in BridgeAppSDK. I can update the comments.

Comment (syoung-smallwisdom) [reply ResearchUXFactory/SBABaseSurveyFactory.swift]: (this is very old code that is designed to be reverse-compatible with AppCore)

**Pull request merged**